### ISSUE TITLE
Submitted transactions aren't immediately visible when calling web3.eth.pendingTransactions

### ISSUE THREAD

**Issue by halfalicious** [ethereum/cpp-ethereum#5008]
If I submit a new transaction and call `web3.eth.pendingTransactions`, I get an empty array:

```
> web3.eth.sendTransaction({from: web3.personal.listAccounts[0], to:'0x884ae368b946a7e5b11462b1333248388b463cc3', value:1000})
"0xe45dbfebd35d94dd2e27c51992aaef298f5e8a76f32e48923520844029ca6923"
> web3.eth.pendingTransactions
[]
```
However, if I perform the same actions in geth, I get the pending transaction:
```
> web3.eth.sendTransaction({from: web3.personal.listAccounts[0], to:'0x884ae368b946a7e5b11462b1333248388b463cc3', value:1000})
"0x157244d1a61cb9c7d935c4ac97a8c0dc666bc907e27b5c31a50286798d4a0eaa"
> web3.eth.pendingTransactions
[{
    blockHash: null,
    blockNumber: null,
    from: "0x1cbb11b72a335b709a61d22bca90f7e54bfcf8b1",
    gas: 90000,
    gasPrice: 11080332409,
    hash: "0x157244d1a61cb9c7d935c4ac97a8c0dc666bc907e27b5c31a50286798d4a0eaa",
    input: "0x",
    nonce: 1,
    r: "0x928f327e22419434186c30567433077f84506d261b23aa1e0401e0f8f7ec4222",
    s: "0x701e1919dec8dc8fa93902f306e439f594e4aff9305c3d246bc4b66dc0a1e515",
    to: "0x884ae368b946a7e5b11462b1333248388b463cc3",
    transactionIndex: 0,
    v: "0x29",
    value: 1000
}]
```
It looks like in the eth case, the transaction is eventually sent out (I confirmed that the tx shows up on [Ropsten Etherscan eventually](https://ropsten.etherscan.io/tx/0xe45dbfebd35d94dd2e27c51992aaef298f5e8a76f32e48923520844029ca6923)). I debugged the code a little bit and confirmed that the tx gets into the transaction queue immediately (`Client::m_tq`), but pending transactions appear to be pulled from a block (`eth::eth_pendingTransactions()` -> `ClientBase::pending()` -> `Client::postSeal()` -> `Block::pending()` -> `return m_transactions`) and I'm not sure when / under what conditions transactions move from the transaction queue to the `Client::m_postSeal` block. 

In any case, I think it would make sense to match geth's behavior in the interest of consistency and (in my humble opinion) more intuitive behavior.

**Comment by gumb0** [comment]
Is eth node synced? I think it doesn't start processing pending transactions until we are close to sync finish

**Comment by gumb0** [comment]
Transferring from TransactionQueue to the pending block should happen (after transaction is validated by the TransactionQueue) in `Client::syncTransactionQueue()` I think

But I don't know all the details of it, it could also not work properly at all. Pending transactions/state is one of the worst test-covered parts

**Comment by halfalicious** [comment]
Yea, the node was pretty up-to-date (within a few hours) but not fully sync'd. I dug through the code a little bit yesterday and I suspect that the transaction queue doesn't get sync'd to the pending block until the local blockchain is completely sync'd (see `Client::resyncStateFromChain`).

I just ran a trial with eth fully sync'd and I'm still getting an empty array. I'll dig a little deeper to see if I can figure out what's going wrong, I managed to see pending transactions when I tested my fix for #4668 so I'm not sure what's different this time. 

On a related note, I think it would make more sense to check the transaction queue (which I suspect is similar to the Bitcoin mempool) rather than the pending block when building the pending transactions array (it looks like transactions aren't immediately removed from the transaction queue after syncing to the pending block which is why I think we can just check the queue rather than both the queue and the block). This must be what geth is doing since it returns a tx that I just submitted despite its local blockchain being very out-of-date. I also did a quick search but didn't find any web3 API documentation for `web3.eth.pendingTransactions `otherwise I'd go off of the publicly spec'd behavior. I will also of course make sure that the behavior of setting a pending filter still works as expected. 

What are your thoughts?

**Comment by gumb0** [comment]
I've asked go-ethereum people and they confirmed that they return transactions of the tx pool sent from the accounts managed by the node https://github.com/ethereum/go-ethereum/blob/6cf0ab38bd0af77d81aad4c104979cebee9e3e63/internal/ethapi/api.go#L1296-L1298

So yeah we should do that, too. I think you'd need to return `m_tq.topTransactions()` from `Client::pending()`. Then `Eth::eth_pendingTransactions()` already does filtering by sender.
